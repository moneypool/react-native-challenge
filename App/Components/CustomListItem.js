import React, { PropTypes } from 'react'
import {
  View,
  TextInput,
  TouchableWithoutFeedback
} from 'react-native'
import * as Animatable from 'react-native-animatable'
import { connect } from 'react-redux'
import Swipeout from 'react-native-swipeout'
import CheckBox from 'react-native-checkbox'
import Styles from './Styles/CustomListItemStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import Colors from '../Themes/Colors'
import ToDosActions from '../Redux/ToDosRedux'

export class CustomListItem extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func,
    taskInfo: PropTypes.shape({
      index: PropTypes.number,
      title: PropTypes.string,
      completed: PropTypes.bool,
      favorite: PropTypes.bool
    }).isRequired,
    toggleCompletedTask: PropTypes.func.isRequired,
    toggleFavorite: PropTypes.func.isRequired,
    removeTask: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    const { title = '' } = this.props.taskInfo || {}

    this.state = {
      editMode: false,
      editText: title
    }
  }

  _onToggleCheckbox (index) {
    this.props.toggleCompletedTask(index)
  }

  _deleteTask (index) {
    this.props.removeTask(index)
  }

  _addFavorite (index) {
    this.props.toggleFavorite(index)
  }

  _toggleEdit () {
    this.setState({ editMode: !this.state.editMode })

    const { editMode } = this.state
    // If buttons are shown, animate in the input. Else animate in the buttons
    if (!editMode) {
      this.refs.cellContainer.lightSpeedOut(250)
        .then(endState => {
          if (endState.finished) {
            // When new state is set, animate in the new container
            this.setState({ editMode: true }, () => {
              this.refs.inputContainer.lightSpeedIn(250)
            })
          }
        })
    } else {
      this.refs.inputContainer.lightSpeedOut(250)
        .then(endState => {
          if (endState.finished) {
            // When new state is set, animate in the new container
            this.setState({ editMode: false }, () => {
              this.refs.cellContainer.lightSpeedIn(250)
            })
          }
        })
    }
  }

  _handleEditTask = (editText) => this.setState({ editText })

  _handlePressEdit = () => {
    const { editText } = this.state
    const { index } = this.props.taskInfo
    this.props.editTask(editText, index)
    this.refs.inputContainer.lightSpeedOut(300)
      .then(endState => {
        if (endState.finished) {
          // When new state is set, animate in the new container
          this.setState({ editMode: false }, () => {
            this.refs.cellContainer.lightSpeedIn(300)
          })
        }
      })
  }

  _renderFavorite (favorite) {
    if (favorite) {
      return (
        <Icon style={Styles.favorite} name='star' />
      )
    } else return null
  }

  _renderEditOrCheckbox () {
    const { editMode, editText } = this.state
    const { index = 0, title = '', completed = true, favorite = true } = this.props.taskInfo || {}
    const swipeBtns = [
      {
        text: 'Favorite',
        backgroundColor: 'gold',
        underlayColor: 'gold',
        onPress: () => this._addFavorite(index)
      },
      {
        text: 'Edit',
        backgroundColor: 'blue',
        underlayColor: 'blue',
        onPress: () => this._toggleEdit()
      },
      {
        text: 'Delete',
        backgroundColor: 'red',
        underlayColor: Colors.error,
        onPress: () => this._deleteTask(index)
      }
    ]
    // When edit mode is off, the normal cell is rendered, with the checkbox
    if (!editMode) {
      return (
        <Animatable.View
          ref='cellContainer'>
          <Swipeout
            right={swipeBtns}
            autoClose
            backgroundColor='transparent'>
            <View>
              <CheckBox
                label={title}
                checked={completed}
                onChange={() => this._onToggleCheckbox(index)}
                containerStyle={Styles.listCheckboxContainer}
                checkboxStyle={Styles.listCheckbox}
                labelStyle={Styles.listText}
              />
              {this._renderFavorite(favorite)}
            </View>
          </Swipeout>
        </Animatable.View>
      )
    } else {
      // Edit mode is on, so render input
      return (
        <Animatable.View
          ref='inputContainer'>
          <View style={Styles.editInput}>
            <TextInput
              style={Styles.editInputText}
              value={editText}
              keyboardType='default'
              returnKeyType='go'
              autoCapitalize='none'
              onChangeText={this._handleEditTask}
              underlineColorAndroid='transparent'
              onSubmitEditing={this._handlePressEdit}
              placeholder='Type to edit task' />
          </View>
          {/* Submit edit button */}
          <TouchableWithoutFeedback onPress={this._handlePressEdit}>
            <Icon style={[Styles.editButtons, Styles.editSubmit]} name='check-circle' />
          </TouchableWithoutFeedback>
          {/* Cancel edit button */}
          <TouchableWithoutFeedback onPress={() => this._toggleEdit()}>
            <Icon style={[Styles.editButtons, Styles.editClose]} name='close' />
          </TouchableWithoutFeedback>
        </Animatable.View>
      )
    }
  }

  render () {
    return (
      <View style={Styles.listContainer}>
        {this._renderEditOrCheckbox()}
      </View>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCompletedTask: index => dispatch(ToDosActions.toggleCompletedTask(index)),
    removeTask: index => dispatch(ToDosActions.removeTask(index)),
    toggleFavorite: index => dispatch(ToDosActions.toggleFavorite(index)),
    editTask: (task, index) => dispatch(ToDosActions.editTask(task, index))
  }
}

export default connect(null, mapDispatchToProps)(CustomListItem)
