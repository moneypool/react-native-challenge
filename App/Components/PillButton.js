import React, { PropTypes } from 'react'
import { TouchableOpacity, Text } from 'react-native'
import Styles from './Styles/PillButtonStyles'

export default class PillButton extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    onPress: PropTypes.func,
    isActive: PropTypes.bool.isRequired
  }

  render () {
    const { isActive } = this.props
    let buttonStyles = [Styles.button]
    let textStyles = [Styles.buttonText]
    if (isActive) {
      buttonStyles.push(Styles.buttonActive)
      textStyles.push(Styles.buttonTextActive)
    }
    return (
      <TouchableOpacity style={buttonStyles} onPress={this.props.onPress}>
        <Text style={textStyles}>{this.props.text}</Text>
      </TouchableOpacity>
    )
  }
}
