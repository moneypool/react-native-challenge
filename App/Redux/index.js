import { combineReducers } from 'redux'
import configureStore from './CreateStore'

export default () => {
    /* ------------- Assemble The Reducers ------------- */
  const rootReducer = combineReducers({
    todos: require('./ToDosRedux').reducer
  })

  return configureStore(rootReducer)
}
