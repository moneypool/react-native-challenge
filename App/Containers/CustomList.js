import React, { PropTypes } from 'react'
import {
  View,
  ScrollView,
  ListView,
  Text
} from 'react-native'
import * as Animatable from 'react-native-animatable'
import { connect } from 'react-redux'
import CustomListItem from '../Components/CustomListItem'
import { filter, sort, toLower } from 'ramda'
import Styles from './Styles/CustomListStyles'
import ToDosActions from '../Redux/ToDosRedux'

export class CustomList extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func,
    tasks: PropTypes.array,
    filterBy: PropTypes.string,
    toggleCompletedTask: PropTypes.func.isRequired,
    removeTask: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    // This could be faster, but checkbox crashes
    const rowHasChanged = (r1, r2) => r1 !== r2
    const ds = new ListView.DataSource({rowHasChanged})

    this.state = {
      dataSource: ds.cloneWithRows([])
    }
  }

  componentWillReceiveProps (nextProps) {
    // Get data from Redux
    const { tasks, filterBy } = nextProps
    // Get dataSource for ListView
    const { dataSource } = this.state
    // Sort data in descending order by title
    let filteredTasks = sort((a, b) => toLower(a.title) > toLower(b.title), tasks)
    // Apply filters if necessary
    if (filterBy === 'Completed') filteredTasks = filter(n => n.completed, filteredTasks)
    if (filterBy === 'Pending') filteredTasks = filter(n => !n.completed, filteredTasks)

    // Update dataSource with completely processed data
    this.setState({dataSource: dataSource.cloneWithRows(filteredTasks)})
  }

  _renderRow (rowData) {
    return (
      <CustomListItem taskInfo={rowData} />
    )
  }

  _renderResults () {
    const { dataSource } = this.state
    const { tasks } = this.props
    if (tasks && tasks.length) {
      return (
        <ListView
          enableEmptySections
          automaticallyAdjustContentInsets={false}
          dataSource={dataSource}
          renderRow={row => this._renderRow(row)}
        />
      )
    } else {
      return (
        <View>
          <Animatable.Text animation='pulse' easing='ease-out' iterationCount='infinite' style={Styles.noResults}>No tasks yet :(</Animatable.Text>
          <Text style={Styles.noResults}>Let's add something fun</Text>
        </View>
      )
    }
  }

  render () {
    return (
      <ScrollView style={Styles.container}>
        {this._renderResults()}
      </ScrollView>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    tasks: state.todos.tasks,
    filterBy: state.todos.filterBy
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCompletedTask: index => dispatch(ToDosActions.toggleCompletedTask(index)),
    removeTask: index => dispatch(ToDosActions.removeTask(index)),
    resetTasks: () => dispatch(ToDosActions.resetTasks()),
    toggleFavorite: index => dispatch(ToDosActions.toggleFavorite(index))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CustomList)
