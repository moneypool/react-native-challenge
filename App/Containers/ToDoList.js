import React, { Component } from 'react'
import {
  View
} from 'react-native'
import Header from './Header'
import CustomList from './CustomList'
import Styles from './Styles/ToDoListStyle'
import { connect } from 'react-redux'
import StartupActions from '../Redux/StartupRedux'
import ReduxPersist from '../Config/ReduxPersist'

export class ToDoList extends Component {
  constructor (props) {
    super(props)

    this.state = {
      newTask: ''
    }
  }

  componentDidMount () {
    // if redux persist is not active fire startup action
    if (!ReduxPersist.active) {
      this.props.startup()
    }
  }

  render () {
    return (
      <View style={Styles.mainContainer} >
        <Header />
        <CustomList />
      </View>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startup: () => dispatch(StartupActions.startup())
})

export default connect(null, mapDispatchToProps)(ToDoList)
