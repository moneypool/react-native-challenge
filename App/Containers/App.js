import React, { Component } from 'react'
import { Provider } from 'react-redux'
import ToDoList from './ToDoList'
import createStore from '../Redux'

// create our store
const store = createStore()

class App extends Component {
  render () {
    return (
      <Provider store={store}>
        <ToDoList />
      </Provider>
    )
  }
}

export default App
