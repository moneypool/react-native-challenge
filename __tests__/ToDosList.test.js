import React from 'react'
import {
  TextInput,
  Text
} from 'react-native'
import { shallow } from 'enzyme'
import { Header } from '../App/Containers/Header'
import { CustomList } from '../App/Containers/CustomList'
import { CustomListItem } from '../App/Components/CustomListItem'
import App from '../App/Containers/App'
import PillButton from '../App/Components/PillButton'
import * as Animatable from 'react-native-animatable'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

// Contains 2 completed tasks, and 1 pending
const tasks = [
  {
    title: 'Take a shower',
    index: 1,
    completed: false,
    favorite: true
  },
  {
    title: 'Drink water',
    index: 2,
    completed: true,
    favorite: false
  },
  {
    title: 'Go to the store',
    index: 3,
    completed: true,
    favorite: false
  }
]

describe('>>>App --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <App />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>Header --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <Header />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>CustomList --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <CustomList />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>CustomListItem --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <CustomListItem />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>Header --- Shallow Render', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Header />)
    wrapper.setProps({ tasks })
  })

  it('render the dumb component', () => {
    expect(wrapper.length).toEqual(1)
  })

  it('contains three buttons when rendered', () => {
    expect(wrapper.find(PillButton)).toHaveLength(3)
  })

  it('contains no buttons when addMode active', () => {
    expect(wrapper.find(PillButton)).toHaveLength(3)
    wrapper.setState({ addMode: true })
    expect(wrapper.find(PillButton)).toHaveLength(0)
  })

  it('contains input addMode', () => {
    wrapper.setState({ addMode: true })
    expect(wrapper.find(TextInput)).toHaveLength(1)
  })

  it('pending button, contains correct pending tasks number', () => {
    expect(wrapper.find(PillButton).at(2).props().text).toBe('Pending - 1')
  })
})

describe('>>>CustomList --- Shallow Render', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<CustomList />)
  })

  it('render the dumb component', () => {
    expect(wrapper.length).toEqual(1)
  })

  it('containt no results text when no tasks are defined', () => {
    expect(wrapper.find(Animatable.Text)).toHaveLength(1)
    expect(wrapper.find(Text)).toHaveLength(1)
    expect(wrapper.find(Animatable.Text).first().props().children).toBe('No tasks yet :(')
    expect(wrapper.find(Text).last().props().children).toBe(`Let's add something fun`)
  })
})

describe('>>>CustomListItem --- Shallow Render', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<CustomListItem />)
  })

  it('render the dumb component', () => {
    expect(wrapper.length).toEqual(1)
  })
})
