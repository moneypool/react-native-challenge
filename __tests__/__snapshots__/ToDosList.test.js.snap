// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>App --- Snapshot render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomColor": "#e32f66",
        "borderBottomWidth": 3,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#161616",
            "fontFamily": "Avenir-Black",
            "fontSize": 18,
            "marginTop": 5,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      myTODOS
    </Text>
    <Text
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#e32f66",
            "fontSize": 24,
            "position": "absolute",
            "right": 10,
            "top": 24,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      testID={undefined}
    >
      î……
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "rgba(179, 233, 199, 0.1)",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#e32f66",
            "borderColor": "rgba(0, 0, 0, 0.0)",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "overflow": "hidden",
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          All
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.0)",
            "borderColor": "#DDD",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Completed
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.0)",
            "borderColor": "#DDD",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Pending - 0
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#D0E0DD",
      }
    }
  >
    <View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e32f66",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginTop": 20,
              "textAlign": "center",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          No tasks yet :(
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e32f66",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginTop": 20,
              "textAlign": "center",
            }
          }
        >
          Let's add something fun
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`>>>CustomList --- Snapshot render correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#D0E0DD",
    }
  }
>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#e32f66",
            "fontSize": 34,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        No tasks yet :(
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#e32f66",
            "fontSize": 34,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Let's add something fun
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`>>>CustomListItem --- Snapshot render correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e32f66",
      "borderBottomWidth": 3,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dbddde",
            "overflow": "hidden",
          },
          undefined,
          Array [
            Object {
              "backgroundColor": "transparent",
            },
          ],
        ]
      }
    >
      <View
        onLayout={[Function]}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {},
            Object {
              "left": 0,
            },
          ]
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginBottom": 5,
                  },
                  Object {
                    "backgroundColor": "#FFF",
                    "margin": 0,
                    "marginBottom": 0,
                    "padding": 10,
                    "paddingRight": 43,
                  },
                ]
              }
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": 26,
                      "width": 26,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "grey",
                        "fontSize": 15,
                      },
                      Object {
                        "fontFamily": "Avenir",
                        "fontSize": 17,
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "gold",
                  "fontSize": 24,
                  "position": "absolute",
                  "right": 5,
                  "top": 10,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î ¸
          </Text>
        </View>
      </View>
      <View />
      <View />
    </View>
  </View>
</View>
`;

exports[`>>>Header --- Snapshot render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottomColor": "#e32f66",
      "borderBottomWidth": 3,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#161616",
          "fontFamily": "Avenir-Black",
          "fontSize": 18,
          "marginTop": 5,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    myTODOS
  </Text>
  <Text
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#e32f66",
          "fontSize": 24,
          "position": "absolute",
          "right": 10,
          "top": 24,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    testID={undefined}
  >
    î……
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(179, 233, 199, 0.1)",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e32f66",
          "borderColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "overflow": "hidden",
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderColor": "#DDD",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        Completed
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderColor": "#DDD",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        Pending - 0
      </Text>
    </View>
  </View>
</View>
`;
